app/models/fact_value.rb

```diff
@@ -81,13 +81,16 @@ def self.count_each(fact, options = {})
     output
   end
 
-  def self.build_facts_hash(facts)
+  def self.build_facts_hash(facts, add_host)
     hosts = Host.where(:id => facts.group_by(&:host_id).keys).all
```

Comment:
This `.where` call is going to be called even when `add_host` is passed, even though it's not used, can you ensure it's only used when needed?